dora/core/common/src/main/java/alluxio/membership/ServiceDiscoveryRecipe.java

```diff
@@ -70,7 +70,7 @@ public ServiceDiscoveryRecipe(AlluxioEtcdClient client, String pathPrefix) {
     mAlluxioEtcdClient = client;
     mRegisterPathPrefix = pathPrefix;
     mExecutor = Executors.newSingleThreadScheduledExecutor(
-        ThreadFactoryUtils.build("service-discovery-checker", false));
+        ThreadFactoryUtils.build("service-discovery-checker", true));
```

Comment:
use daemon thread so the existence of the thread does not block process exit
https://stackoverflow.com/questions/2213340/what-is-a-daemon-thread-in-java